tonic/Cargo.toml

```diff
@@ -28,10 +28,13 @@ gzip = ["dep:flate2"]
 zstd = ["dep:zstd"]
 default = ["transport", "codegen", "prost"]
 prost = ["dep:prost"]
-tls = ["dep:rustls-pemfile", "dep:tokio-rustls", "dep:tokio", "tokio?/rt", "tokio?/macros"]
+tls = ["tls-any", "tls-ring"] # Deprecated. Please use `tls-ring` or `tls-aws-lc` instead.
+tls-any = ["dep:rustls-pemfile", "dep:tokio-rustls", "dep:tokio", "tokio?/rt", "tokio?/macros"]
+tls-ring = ["tls-any", "tokio-rustls/ring"]
+tls-aws-lc = ["tls-any", "tokio-rustls/aws-lc-rs"]
 tls-roots = ["tls-native-roots"] # Deprecated. Please use `tls-native-roots` instead.
-tls-native-roots = ["tls", "channel", "dep:rustls-native-certs"]
-tls-webpki-roots = ["tls", "channel", "dep:webpki-roots"]
+tls-native-roots = ["channel", "dep:rustls-native-certs"]
+tls-webpki-roots = ["channel", "dep:webpki-roots"]
```

Comment:
This seems to make a breaking change.

Reply:
this will break users who depend on tonic like
```toml
tonic = { version = "0.12", features = ["tls-webpki-roots"]}
```
to imply the `tls` feature. I thought this way was better than creating separate `tls-aws-lc-webpki-roots` and `tls-aws-lc-native-roots` features. Open to change here

Reply:
We just merged some breaking changes, so we should probably do what's best without regards for backwards compatibility at this point.

Reply:
As another breaking change was merged after my comment, I opened #2014 to confirm if it's okay to make other changes. I think the proposal removing `tls` from these features is good when we can make breaking changes.